### PR TITLE
refactor(ivy): undeprecate inject

### PR DESCRIPTION
**Pull request by benlesh**
Per offline discussion, `inject` is already an API that is seeing limited use. We are going to undeprecate it _for now_.

### CODE REVIEW

File: packages/core/src/di/injector_compatibility.ts [hunk 1]
```diff
@@ -104,8 +104,6 @@ export function ɵɵinject<T>(token: Type<T>| InjectionToken<T>, flags = InjectF
   return (_injectImplementation || injectInjectorOnly)(token, flags);
 }
 /**
- * @deprecated in v8, delete after v10. This API should be used only be generated code, and that
- * code should now use ɵɵinject instead.
```
Comment (kara) [inline packages/core/src/di/injector_compatibility.ts]: We should also remove the warning above

```
 WARNING: This function is meant to be generated by the Ivy compiler, and is not meant for
 * developer consumption!	 * developer consumption!
```

File: packages/core/src/di/injector_compatibility.ts [hunk 2]
```diff
@@ -104,8 +104,6 @@ export function ɵɵinject<T>(token: Type<T>| InjectionToken<T>, flags = InjectF
   return (_injectImplementation || injectInjectorOnly)(token, flags);
```
Comment (kara) [inline packages/core/src/di/injector_compatibility.ts]: We should also remove the prefix

**Review by kara — CHANGES_REQUESTED** [review]

File: packages/core/src/di/injector_compatibility.ts (hunk 2 above)
Comment (mhevery) [reply packages/core/src/di/injector_compatibility.ts]: I think we should leave it as is. This way we have it exported twice once for public consumption and once for the generated code.

**Review by kara — APPROVED** [review]
LGTM

File: packages/core/src/di/injector_compatibility.ts (hunk 2 above)
Comment (kara) [reply packages/core/src/di/injector_compatibility.ts]: SGTM

**Comment by benlesh** [comment]
[presubmit](http://test/OCL:245336696:BASE:245336721:1556237589255:d57c564a)

**Review by mhevery — APPROVED** [review]

**Review by IgorMinar — APPROVED** [review]

**Pull request closed without merging**